mlflow/utils/conda.py

```diff
@@ -126,36 +131,55 @@ def get_or_create_conda_env(conda_env_path, env_id=None):
             )
         )
 
-    (_, stdout, _) = process.exec_cmd([conda_path, "env", "list", "--json"])
-    env_names = [os.path.basename(env) for env in json.loads(stdout)["envs"]]
+    env_names = _list_conda_environments()
     project_env_name = _get_conda_env_name(conda_env_path, env_id)
     if project_env_name not in env_names:
         _logger.info("=== Creating conda environment %s ===", project_env_name)
-        if conda_env_path:
-            process.exec_cmd(
-                [
-                    conda_env_create_path,
-                    "env",
-                    "create",
-                    "-n",
+        try:
+            if conda_env_path:
+                process.exec_cmd(
+                    [
+                        conda_env_create_path,
+                        "env",
+                        "create",
+                        "-n",
+                        project_env_name,
+                        "--file",
+                        conda_env_path,
```

Comment:
Question:
Shall we add `"--yes"` argument for `conda env create` command ?

Reply:
Added `--yes` to `conda create` for safety.

Reply:
Without `--yes`, `conda create` does ask for confirmation but it appears `subprocess` automatically accepts it (not 100% sure though).

Reply:
Can confirm. 

``` bash
~/Documents via 🅒 base
➜ conda env list
# conda environments:
#
base                  *  /Users/benwilson/opt/anaconda3
diviner-dev              /Users/benwilson/opt/anaconda3/envs/diviner-dev
hodor                    /Users/benwilson/opt/anaconda3/envs/hodor
mlflow-dev-env           /Users/benwilson/opt/anaconda3/envs/mlflow-dev-env

(base)
~/Documents via 🅒 base took 2s
➜ (cat conda.yaml && conda env remove -n hodor)
name: hodor
channels:
  - conda-forge
dependencies:
  - python=3.9
  - pip
  - pip:
    - mlflow==1.20.0

Remove all packages in environment /Users/benwilson/opt/anaconda3/envs/hodor:

(base)
```
It subprocesses auto-accept :)